### PR TITLE
[search] removing SearchApiKeyCredential and using AzureKeyCredential from core

### LINKED ISSUE #10530: [search] take AzureKeyCredential from core

### PR DESCRIPTION
**Pull request by iscai-msft**
rebased off #10509
fixes #10530

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/core/azure-core/azure/core/credentials.py [hunk 1]
```diff
@@ -28,3 +29,45 @@ def get_token(self, *scopes, **kwargs):
     from collections import namedtuple
 
     AccessToken = namedtuple("AccessToken", ["token", "expires_on"])
+
+__all__ = ["AzureKeyCredential"]
+
+
+class AzureKeyCredential(object):
+    """Credential type used for authenticating to an Azure service.
+    It provides the ability to update the key without creating a new client.
```
Comment (xiangyan99) [inline sdk/core/azure-core/azure/core/credentials.py]: What is the scenario that user needs to update key after creation?
And do we want to support per-request key? (get key from context in on_request method)

Comment (iscai-msft) [reply sdk/core/azure-core/azure/core/credentials.py]: we want to support key rolling / rotating, which is why we have the ability to update keys.

I don't think there're any plans for adding a per-request key, with this pr i'm just trying to move the key credential we already had in place for search and text analytics into azure core for shared usage

File: sdk/search/azure-search-documents/azure/search/documents/_credential.py [hunk 2]
```diff
@@ -6,51 +6,15 @@
...
-        """Update the API key.
-        This can be used when you've regenerated your service API key and want
-        to update long-lived clients.
-
-        :param str key: The API key to your Azure search account.
-        """
-        self._api_key = key
-
-
 class HeadersMixin(object):
```
Comment (xiangyan99) [inline sdk/search/azure-search-documents/azure/search/documents/_credential.py]: Given there is no credential at all in this file, should we rename it?

**Pull request merged**